### ISSUE TITLE
关于微调reranker-v2-m3系列模型的数据集问题

### ISSUE THREAD

**Issue by Zhouziyi828** [FlagOpen/FlagEmbedding#880]
我观察到在您给的微调数据集例子中,pos只有一条.在您关于其他问题的回复中我看到:
fine-tune的数据集中可以包含多条正样本，每次过到该数据时候会随机从正样本列表中选取一条作为训练，所以正样本的数目差异很大目前对训练影响不大。

有以下问题:
1. 目前我们无法使用多个正样本么?
2. 我们想做RAG场景下rerank的微调,如果我们构造的数据集中的正样本是有相关性顺序的,类似t2Ranking这个数据集中,pos的构造是从相关性最高到最低的,这样的数据集对我们的微调有帮助么?

**Comment by staoxiao** [comment]
1. The current code only supports one positive within a training step. Since the code will sample different positives for different steps, multiple positives will still be used when training more than one epoch.
2. If you want to train the model with examples that have different relevance scores, you need to change the loss function (e.g, ranknet loss function): https://github.com/FlagOpen/FlagEmbedding/blob/master/FlagEmbedding/baai_general_embedding/finetune/modeling.py#L101. 
We will consider adding more loss functions to this tool in the future.

**Comment by Zhouziyi828** [comment]
谢谢您的回复.如果我们修改这样的损失函数,将不同的pos考虑进去,那我们就需要在每一个training step使用多个pos,可以修改么?这部分代码在哪里修改呢?
